scripts/buildWebtypes.js

```diff
@@ -0,0 +1,235 @@
+      },
+    }));
+  const events = elementAnalysis.events.map((event) => ({
+    name: event.name,
+    description: event.description,
+  }));
+
+  return {
+    name: elementAnalysis.tagname,
+    description: elementAnalysis.description,
+    attributes,
+    js: {
+      properties,
+      events,
+    },
+  };
+}
+
+function createPlainWebTypes(packageJson, packageElements) {
+  return {
+    $schema: 'https://json.schemastore.org/web-types',
+    name: packageJson.name,
+    version: packageJson.version,
+    'description-markup': 'markdown',
+    contributions: {
+      html: {
+        elements: packageElements.map(createPlainElementDefinition),
+      },
+    },
+  };
+}
+
+function createLitElementDefinition(elementAnalysis) {
+  const booleanAttributes = elementAnalysis.properties
+    .filter((prop) => prop.privacy === 'public')
+    .filter((prop) => !prop.metadata.polymer.readOnly)
+    .filter((prop) => prop.type.includes('boolean'))
+    .map((prop) => ({
+      name: `?${prop.name}`,
+      description: prop.description,
+      value: {
+        // Type checking does not work with template tagged literals
+        // Since this Lit binding should use an expression, just declare it as such
+        kind: 'expression',
+      },
+    }));
+  const propertyAttributes = elementAnalysis.properties
+    .filter((prop) => prop.privacy === 'public')
+    .filter((prop) => !prop.metadata.polymer.readOnly)
+    .filter((prop) => !prop.type.includes('boolean'))
+    .map((prop) => ({
+      name: `.${prop.name}`,
+      description: prop.description,
+      value: {
+        // Type checking does not work with template tagged literals
+        // Since this Lit binding should use an expression, just declare it as such
+        kind: 'expression',
+      },
+    }));
+  const eventAttributes = elementAnalysis.events.map((event) => ({
+    name: `@${event.name}`,
+    description: event.description,
+    value: {
+      // Type checking does not work with template tagged literals
+      // Since this Lit binding should use an expression, just declare it as such
+      kind: 'expression',
+    },
+  }));
+
+  return {
+    name: elementAnalysis.tagname,
+    description: elementAnalysis.description,
+    // Declare as extension to plain web type, this also means we don't have to
+    // repeat the same stuff from the plain web-types.json again
+    extension: true,
+    // IntelliJ does not understand Lit template syntax, so
+    // effectively everything has to be declared as attribute
+    attributes: [...booleanAttributes, ...propertyAttributes, ...eventAttributes],
+  };
+}
+
+function createLitWebTypes(packageJson, packageElements) {
+  return {
+    $schema: 'https://json.schemastore.org/web-types',
+    name: packageJson.name,
+    version: packageJson.version,
+    'description-markup': 'markdown',
+    framework: 'lit',
+    'framework-config': {
+      'enable-when': {
+        'node-packages': ['lit'],
```

Comment:
The Lit-specific definitions will only be used if the `lit` package is installed. Though that seems to happen when you install any of our components, so yeah, doesn't make much of a difference at the moment.